pkg/squashfs/squashfs.go

```diff
@@ -211,12 +254,27 @@ var squashNotFound = errors.Errorf("squashfuse program not found")
 // return a pointer to the squashfuse cmd.
 // The caller of the this is responsible for the process created.
 func squashFuse(squashFile, extractDir string) (*exec.Cmd, error) {
-	sqfuse := findSquashfusePath()
 	var cmd *exec.Cmd
-	if sqfuse == "" {
+
+	once.Do(findSquashFuseInfo)
+	if squashFuseInfo.Path == "" {
 		return cmd, squashNotFound
 	}
 
+	notifyOpts := ""
+	notifyPath := ""
+	if squashFuseInfo.SupportsNotfiy {
+		sockdir, err := os.MkdirTemp("", "sock")
```

Comment:
What's going to clean this directory up?

Can we do a "defer os.RemoveAll(sockdir)"?  After this function returns, the socket is no longer needed, right?

Reply:
Sorry, I'm not used to cleaning up resourcing manually. I will add the defer.

Reply:
"How barbaric!"

https://www.youtube.com/watch?v=LkqiDu1BQXY